src/mlpack/tests/cv_test.cpp

```diff
@@ -74,6 +75,81 @@ TEST_CASE("BinaryClassificationMetricsTest", "[CVTest]")
 
   double f1 = 2 * 0.6 * 0.75 / (0.6 + 0.75);
   REQUIRE(F1<Binary>::Evaluate(lr, data, labels) == Approx(f1).epsilon(1e-7));
+
+
+  // Testing binary ROC-AUC Score.
+  arma::Row<size_t> rocTrueLabels;
+  arma::Row<double> rocScoresOfPC;
+  double rocAucScore;
+
+  // Test - 1
+  rocTrueLabels = arma::Row<size_t>("0 0 0 0 0  1 1 1 1 1");
+  rocScoresOfPC = arma::Row<double>("0.1 0.2 0.3 0.4 0.5  0.6 0.7 0.8 0.9 1");
+
+  rocAucScore = 0;
```

Comment:
Is this correct?  If the true class is 0, then I would expect the ROC-AUC to be nonzero, since it is possible to choose a threshold such that some points are correctly classified.

Reply:
Yes. Even though some points are classified correctly, we may get ROC AUC score as zero.
<br/>
**For example**:
```
TrueLabels = [0, 0, 0, 0, 1, 1, 1, 1]
ScoresOfPC = [.4, .5, .6, .7, .8, .85, .9, .95]
```
If we consider `0` as PositiveClass, we will get the predicted labels as `[1, 1, 0, 0, 0, 0, 0, 0]`, so the outcomes are `[FN, FN, TP, TP, FP, FP, FP, FP]`. In this case we can see that there are some TruePositives, but still we will get ROC AUC score as zero.

The reason is that, all the FalsePositives have taken the highest possible threshold values. So by the time we reach the threshold `0.7`(since we are considering the threshold values in descending order), the FalsePositiveRate is already `1` and TruePositiveRate is still `0`. So the ROC curve will be flat and the AUC score is zero.